webhook/webhook.go

```diff
@@ -35,5 +42,43 @@ func main() {
 }
 
 func handleGithubEvent(w http.ResponseWriter, r *http.Request) {
-	// TODO (priyawadhwa@): Add logic to handle a Github event here
+	eventType := r.Header.Get("X-GitHub-Event")
+	if eventType != constants.PullRequestEvent {
+		return
+	}
+	event := new(github.PullRequestEvent)
+	if err := json.NewDecoder(r.Body).Decode(event); err != nil {
+		log.Printf("error decoding pr event: %v", err)
+	}
+	if err := handlePullRequestEvent(event); err != nil {
+		log.Printf("error handling pr event: %v", err)
+	}
+}
+
+func handlePullRequestEvent(event *github.PullRequestEvent) error {
+	// Cleanup any deployments if PR was merged or closed
+	if event.GetAction() == constants.ClosedAction {
+		return kubernetes.CleanupDeployment(event)
+	}
+
+	// Only continue if the docs-modifications label was added
+	if event.GetAction() != constants.LabeledAction {
+		return nil
+	}
+
+	if event.GetPullRequest() == nil {
```

Comment:
is this possible on a `PullRequestEvent`?

Reply:
Just read the docs and it doesn't seem like it, I'll remove this check as well